internal/auth0/guardian/flatten.go

```diff
@@ -0,0 +1,202 @@
+
+	twilio, err := api.Guardian.MultiFactor.SMS.Twilio()
+	if err != nil {
+		return nil, err
+	}
+
+	m["auth_token"] = twilio.GetAuthToken()
+	m["from"] = twilio.GetFrom()
+	m["messaging_service_sid"] = twilio.GetMessagingServiceSid()
+	m["sid"] = twilio.GetSID()
+
+	return []interface{}{m}, nil
+}
+
+func flattenWebAuthnRoaming(enabled bool, api *management.Management) ([]interface{}, error) {
+	webAuthnRoamingData := make(map[string]interface{})
+	webAuthnRoamingData["enabled"] = enabled
+
+	if !enabled {
+		return []interface{}{webAuthnRoamingData}, nil
+	}
+
+	webAuthnSettings, err := api.Guardian.MultiFactor.WebAuthnRoaming.Read()
+	if err != nil {
+		return nil, err
+	}
+
+	webAuthnRoamingData["user_verification"] = webAuthnSettings.GetUserVerification()
+	webAuthnRoamingData["override_relying_party"] = webAuthnSettings.GetOverrideRelyingParty()
+	webAuthnRoamingData["relying_party_identifier"] = webAuthnSettings.GetRelyingPartyIdentifier()
+
+	return []interface{}{webAuthnRoamingData}, nil
+}
+
+func flattenWebAuthnPlatform(enabled bool, api *management.Management) ([]interface{}, error) {
+	webAuthnPlatformData := make(map[string]interface{})
+	webAuthnPlatformData["enabled"] = enabled
+
+	if !enabled {
+		return []interface{}{webAuthnPlatformData}, nil
+	}
+
+	webAuthnSettings, err := api.Guardian.MultiFactor.WebAuthnPlatform.Read()
+	if err != nil {
+		return nil, err
+	}
+
+	webAuthnPlatformData["override_relying_party"] = webAuthnSettings.GetOverrideRelyingParty()
+	webAuthnPlatformData["relying_party_identifier"] = webAuthnSettings.GetRelyingPartyIdentifier()
+
+	return []interface{}{webAuthnPlatformData}, nil
+}
+
+func flattenDUO(enabled bool, api *management.Management) ([]interface{}, error) {
+	duoData := make(map[string]interface{})
+	duoData["enabled"] = enabled
+
+	if !enabled {
+		return []interface{}{duoData}, nil
+	}
+
+	duoSettings, err := api.Guardian.MultiFactor.DUO.Read()
+	if err != nil {
+		return nil, err
+	}
+
+	duoData["integration_key"] = duoSettings.GetIntegrationKey()
+	duoData["secret_key"] = duoSettings.GetSecretKey()
+	duoData["hostname"] = duoSettings.GetHostname()
+
+	return []interface{}{duoData}, nil
+}
+
+func flattenPush(d *schema.ResourceData, enabled bool, api *management.Management) ([]interface{}, error) {
+	pushData := make(map[string]interface{})
+	pushData["enabled"] = enabled
+
+	if !enabled {
+		return []interface{}{pushData}, nil
+	}
+
+	pushProvider, err := api.Guardian.MultiFactor.Push.Provider()
+	if err != nil {
+		return nil, err
+	}
+	pushData["provider"] = pushProvider.GetProvider()
+
+	customApp, err := api.Guardian.MultiFactor.Push.CustomApp()
+	if err != nil {
+		return nil, err
+	}
+
+	pushData["custom_app"] = []interface{}{
+		map[string]interface{}{
+			"app_name":        customApp.GetAppName(),
+			"apple_app_link":  customApp.GetAppleAppLink(),
+			"google_app_link": customApp.GetGoogleAppLink(),
+		},
+	}
+
+	if pushProvider.GetProvider() == "sns" {
```

Comment:
SNS is not supported yet by this terraform provider, right? https://auth0.com/docs/api/management/v2#!/Guardian/get_apns

Reply:
Sadly not yet :) But hopefully soon!